frame/support/procedural/src/lib.rs

```diff
@@ -1738,3 +1738,44 @@ pub fn origin(_: TokenStream, _: TokenStream) -> TokenStream {
 pub fn composite_enum(_: TokenStream, _: TokenStream) -> TokenStream {
 	pallet_macro_stub()
 }
+
+/// An attribute macro that can be attached to a module declaration. Doing so will
+/// make it available for import later.
+#[proc_macro_attribute]
+pub fn export_section(attr: TokenStream, tokens: TokenStream) -> TokenStream {
+	match macro_magic::mm_core::export_tokens_internal(attr, tokens, false) {
+		Ok(tokens) => tokens.into(),
+		Err(err) => err.to_compile_error().into(),
+	}
+}
+
+/// An attribute macro that can be attached to a module declaration. Doing so will
+/// import the content of the section that was exported previously using [`macro@export_section`].
+#[import_tokens_attr]
+#[proc_macro_attribute]
+pub fn import_section(attr: TokenStream, tokens: TokenStream) -> TokenStream {
+	let foreign_mod = parse_macro_input!(attr as ItemMod);
+	let mut internal_mod = parse_macro_input!(tokens as ItemMod);
+
+	let err = syn::Error::new_spanned(
+		internal_mod.clone(),
+		"A section can only be imported under a pallet macro",
+	)
+	.to_compile_error()
+	.into();
+
+	if internal_mod.attrs.len() == 0 {
+		return err
+	}
```

Comment:
This should do it and should also check that the `foreign_mod` is valid as well (should have `#[pallet_section]` attached to it, assuming you take my suggestion of renaming to `#[pallet_section]`.

```suggestion
	// check that internal_mod is a pallet module
	if internal_mod.attrs.iter().find(|attr| {
	    if let Some(last_seg) = attr.path().segments.last() {
	        last_seg.ident == "pallet"
	    } else {
	        false
	    }
	}).is_none() {
	    return Err(Error::new(
	        internal_mod.span(),
	        "`#[import_section]` can only be applied to a valid pallet module",
	    )).to_compile_error.into();
	}

    // check that foreign_mod has `#[pallet_section]` attached to it
	if foreign_mod.attrs.iter().find(|attr| {
	    if let Some(last_seg) = attr.path().segments.last() {
	        last_seg.ident == "pallet_section"
	    } else {
	        false
	    }
	}).is_none() {
	    return Err(Error::new(
	        foreign_mod.span(),
	        "`#[import_section]` can only take a valid pallet section as an argument",
	    )).to_compile_error().into();
	}
```